inc/connector/namespace.php

```diff
@@ -382,16 +382,39 @@ function pre_get_scheduled_event( $pre, $hook, $args, $timestamp ) {
 	}
 
 	$jobs = Job::get_jobs_by_query( [
+		'args' => null,
+		'hook' => null,
+		'limit' => 100,
+	] );
+
+	$filter = [
 		'hook' => $hook,
 		'timestamp' => $timestamp,
 		'args' => $args,
-	] );
+	];
+
+	// Filter results array.
+	$jobs = array_filter( $jobs, function ( $event ) use ( $filter ) : bool {
+		if ( is_string( $filter['hook'] ) && $event->hook !== $filter['hook'] ) {
+			return false;
+		}
+
+		if ( ! is_null( $filter['args'] ) && $event->args !== $filter['args'] ) {
+			return false;
+		}
+
+		if ( is_int( $filter['timestamp'] ) ) {
+			return $event->nextrun === $filter['timestamp'];
+		}
+
+		return true;
+	} );
 
 	if ( is_wp_error( $jobs ) || empty( $jobs ) ) {
 		return false;
 	}
 
-	$job = $jobs[0];
+	$job = array_shift( $jobs );
```

Comment:
`$jobs` may be a `WP_Error` instance, so passing that to `array_filter` will throw a Fatal error.

Move the first part of the check from line 413 to be right after querying the jobs.

That said, given we are only interested in a single job, there is no point in processing all 100. Let's stop with the first job that matches our criteria.

Overall, this could look like so:

```php
	if ( is_wp_error( $jobs ) ) {
		return false;
	}

	$find_job = function ( array $jobs, array $filter ) {
		foreach ( $jobs as $job ) {
			if ( is_string( $filter['hook'] ) && $event->hook !== $filter['hook'] ) {
				continue;
			}

			if ( ! is_null( $filter['args'] ) && $event->args !== $filter['args'] ) {
				continue;
			}

			if ( is_int( $filter['timestamp'] ) && $event->nextrun !== $filter['timestamp'] ) {
				continue;
			}

			return $job;
		}

		return null;
	};

	$job = $find_job( $jobs, compact( 'hook', 'timestamp', 'args' ) );

	if ( ! $job ) {
		return false;
	}
```

Please note that this is untested code. 😬

Reply:
Good idea stopping after the first match, nice one!